src/rooms/schema/RoomState.ts

```diff
@@ -104,13 +110,19 @@ export class RoomState extends Schema {
     return array2D;
   }
 
-  isWalkable(x: number, y: number): boolean {
+  isWalkableForPlayer(x: number, y: number): boolean {
     if (x < 0 || x >= this.width || y < 0 || y >= this.height) {
       return false;
     }
     return this.getCellValue(x, y) === 0;
   }
 
+  isWalkableForCrate(x: number, y: number): boolean {
+    if (x < 0 || x >= this.width || y < 0 || y >= this.height) return false;
+    const cell = this.getCellValue(x, y);
+    return cell === 0 || cell === 2; 
+  }
```

Comment:
Currently crate can be pushed to the cell with a player on it.
```suggestion
  isWalkableForCrate(x: number, y: number): boolean {
    if (x < 0 || x >= this.width || y < 0 || y >= this.height) return false;
    const cell = this.getCellValue(x, y);
    // check if player occupies
    return cell === 0 || cell === 2; 
  }
```